File: src/Mob_ChuChu.java

public class Mob_ChuChu extends Mob{
	
	private static final int damage = 3;
	private static final int health = 5;
	private int speedmod = 4;
	private int tic = 0;
	
	Mob_ChuChu(int x, int y){
		super("ChuChu", damage, health, x, y);
		this.sprite = getImage("graphics/Sprout01.png");
	}
	
	Mob_ChuChu(int damage, int health, int x, int y){
		super("ChuChu", damage, health, x, y);
		this.sprite = getImage("graphics/Sprout01.png");
	}
	
	@Override
	public void pattern()
	{
			jump();
			if(tic>100){
				speedmod = -speedmod;
				tic = 0;
			}
			setXspeed(speedmod);
			move();
			tic++;
	}
}

File: src/mob/MobChuChu.kt
package mob

import resources.graphics.ImageUtils

class MobChuChu(
        name: String,
        damage: Int,
        health: Int,
        x: Int,
        y: Int
) : Mob(name, damage, health, x, y) {

    private var speedmod = 4
    private var tic = 0

    init {
        this.sprite = ImageUtils.getImage("Sprout01.png")
    }

    internal constructor(x: Int, y: Int) : this("ChuChu", damage, 5, x, y)
    internal constructor(damage: Int, health: Int, x: Int, y: Int) : this("ChuChu", damage, health, x, y)

    override fun pattern() {
        jump()
        if (tic > 100) {
            speedmod = -speedmod
            tic = 0
        }
        xspeed = speedmod
        move()
        tic++
    }

    companion object {
        private val damage = 3
        private val health = 5
    }
}
